### ISSUE TITLE
Screw up reshape and merge: Getting "theano.function was asked to create a function computing outputs given certain inputs, but the provided input variable at index 2 is not part of the computational graph needed to compute the outputs: lambda_1_target."

### ISSUE THREAD

**Issue by wailoktam** [fchollet/keras#2777]
Hi, I badly need help on this. I suppose it is a problem of the network structure. Below is the output of model.summary:

---
# Layer (type)                       Output Shape        Param #     Connected to                     

dense_1 (Dense)                    (None, 200)         2000200     dense_input_1[0][0]              

---

reshape_1 (Reshape)                (None, 1, 200)      0           dense_1[0][0]                    

---

convolution1d_1 (Convolution1D)    (None, 1, 10)       6010        reshape_1[0][0]                  

---

activation_1 (Activation)          (None, 1, 10)       0           convolution1d_1[0][0]            

---

dense_2 (Dense)                    (None, 200)         2000200     dense_input_2[0][0]              

---

reshape_2 (Reshape)                (None, 1, 200)      0           dense_2[0][0]                    

---

convolution1d_2 (Convolution1D)    (None, 1, 10)       6010        reshape_2[0][0]                  

---

activation_2 (Activation)          (None, 1, 10)       0           convolution1d_2[0][0]            

---
# lambda_1 (Lambda)                  (None, 1)           0           merge_1[0][0]                    

Initially, I have a maxpooling 1d layer after each of the two relu layer but now I comment them out. If I put them back, I get a floating point exception after the warning in the title. Now I can complete execution without the maxpooling but the loss is always NAN. I actually need the Maxpooling layer to be there because it is said to be there by some paper that I need to follow during implementation. 

A billion thanks in advance. 

Please make sure that the boxes below are checked before you submit your issue. Thank you!
- [ O ] Check that you are up-to-date with the master branch of Keras. You can update with:
  pip install git+git://github.com/fchollet/keras.git --upgrade --no-deps
- [ O ] If running on Theano, check that you are up-to-date with the master branch of Theano. You can update with:
  pip install git+git://github.com/Theano/Theano.git --upgrade --no-deps
- [ O ] Provide a link to a GitHub Gist of a Python script that can reproduce your issue (or just copy the script here if it is short).

**Comment by phdowling** [comment]
You should probably reformat your question a bit, it's hard to tell what's going on exactly. Maybe show us the code of how you construct the model as well!

**Comment by wailoktam** [comment]
Hi, thanks for responding. Below is my code:

    leftKerasModel = Sequential()
    leftKerasModel.add(Reshape((1,100,100),  input_shape=(100, 100)))
    leftKerasModel.add(Convolution2D(10, 3, 3, border_mode='same'))
    leftKerasModel.add(Activation('relu'))
    leftKerasModel.add(MaxPooling2D(pool_size=(2, 2)))

    rightKerasModel = Sequential()
    rightKerasModel.add(Reshape((1,100,100), input_shape=(100,100)))
    rightKerasModel.add(Convolution2D(10, 3, 3, border_mode='same'))
    rightKerasModel.add(Activation('relu'))
    rightKerasModel.add(MaxPooling2D(pool_size=(2, 2)))

    mergedKerasModel = Sequential()
    merged = Merge([leftKerasModel, rightKerasModel], mode='cos', dot_axis=1)
    mergedKerasModel.add(merged)
    mergedKerasModel.add(Lambda(lambda x: 1-x))

    sgd = SGD(lr=0.01, decay=1e-6, momentum=0.9, nesterov=True)
    mergedKerasModel.compile(loss='custom_objective', optimizer=sgd)
    mergedKerasModel.fit([test3dLArray, test3dRArray], testLabels, nb_epoch=10, batch_size=32)

If I do without the lambda layer, the same complaint is made against the merge layer. It seems that something is wrong with merge in dot or cos mode.
